ccxt.js

```diff
@@ -13161,26 +13193,25 @@ var poloniex = {
     },
 
     async fetchMyTrades (market = undefined, params = {}) {
-        let now = this.seconds ();
+        let m = undefined;
+        if (market)
+            m = this.market (market);
+        let pair = m ? m['id'] : 'all';
         let request = this.extend ({
-            'currencyPair': 'all',
+            'currencyPair': pair,
             'end': this.seconds (), // last 50000 trades by default
         }, params);
-        if (market) {
-            let m = this.market (market);
-            request['currencyPair'] = m['id'];
-        }
         let trades = await this.privatePostReturnTradeHistory (request);
-        if (market)
+        if (market) // This is a hard to read control flow
             return this.parseTrades (trades, m);
         let result = { 'info': trades };
         let ids = Object.keys (trades);
         for (let i = 0; i < ids.length; i++) {
             let id = ids[i];
-            let trades = trades[id];
-            let market = this.markets_by_id[id];
+            let trades = trades[id]; // This is a misleading variable name
+            let market = this.markets_by_id[id]; // This is a misleading varialbe name
```

Comment:
No, 'market', because it already exists and should be undefined here because of line 13205